internal/cmd/schema.go

```diff
@@ -360,22 +363,38 @@ func determinePrefixForSchema(ctx context.Context, specifiedPrefix string, clien
 	return "", nil
 }
 
-// Compiles an input schema written in the new composable schema syntax
-// and produces it as a fully-realized schema
-func schemaCompileCmdFunc(cmd *cobra.Command, args []string, termChecker termChecker) error {
-	stdOutFd, err := safecast.Convert[int](os.Stdout.Fd())
-	if err != nil {
-		return err
-	}
+func schemaCompileOuter(cmd *cobra.Command, args []string) (bool, error) {
 	outputFilepath := cobrautil.MustGetString(cmd, "out")
-	if outputFilepath == "" && !termChecker.IsTerminal(stdOutFd) {
-		return errors.New("must provide stdout or output file path")
```

Comment:
If we don't have this do we get weird behavior in environments that don't have stdout available? Or is that captured by the error behavior of writing to `os.Stdout`?

Reply:
I think it's captured by the call to `fmt.Fprint`.